verification/cbmc/proofs/aws_byte_cursor_advance_nospec/aws_byte_cursor_advance_nospec_harness.c

```diff
@@ -36,7 +36,9 @@ void aws_byte_cursor_advance_nospec_harness() {
     } else {
         assert(rv.ptr == old.ptr);
         assert(rv.len == len);
-        assert(cursor.ptr == old.ptr + len);
+        if (old.ptr) {
+            assert(cursor.ptr == old.ptr + len);
+        }
```

Comment:
Use `AWS_IMPLIES`? See first example.

Reply:
Hmm, it seems to me like we are losing something here after this change.
Now we don't seem to check anything if `old.ptr` is `NULL`.

How about:
```c
if (old_cur.ptr) {
    assert(cursor.ptr == old.ptr + len);
} else {
    assert(cursor.ptr == NULL);
}
```